libs/blocks/graybox-promote/components/graybox-promote-lit.js

```diff
@@ -48,80 +47,87 @@ const getSheetValue = (data, key) => data?.find(
   (obj) => obj.key?.toLowerCase() === key?.toLowerCase(),
 )?.value;
 
-const getAemInfo = () => {
+const getAemInfo = (context) => {
   const search = new URLSearchParams(window.location.search);
-  return {
-    ref: search.get('ref'),
-    repo: search.get('repo'),
-    owner: search.get('owner'),
-    excelRef: search.get('referrer'),
-  };
+  context.ref = search.get('ref');
+  context.repo = search.get('repo');
+  context.owner = search.get('owner');
+  context.excelRef = search.get('referrer');
 };
 
-const getProjectInfo = async (url) => {
-  const liveOrigin = url.origin.replace('.hlx.page', '.hlx.live');
+const getProjectInfo = async (context) => {
   const sheet = await getJson(
-    `${liveOrigin}${url.pathname}?sheet=settings`,
+    `${context.liveUrl.href}?sheet=settings`,
     'Failed to fetch project info',
   );
-  return {
-    experienceName: getSheetValue(
-      sheet.data,
-      KEYS.PROJECT_INFO.EXPERIENCE_NAME,
-    ),
-    grayboxIoEnv: getSheetValue(sheet.data, KEYS.PROJECT_INFO.GRAYBOX_IO_ENV),
-  };
+  context.setup.experienceName = getSheetValue(
```

Comment:
we should make sure context.setup is defined, I don't see that anywhere.

Reply:
The context.setup is initialised in [here](https://github.com/adobecom/milo/pull/2584/files#diff-7214629f44f7a1dbc0bdfc18fb6309965da48000fe961e4aaeed56046e9aa06fR129). Later on I pass the `this`  instance to the helper functions and I refer to it as `context` inside those functions.